docker/build-tools/Dockerfile

```diff
@@ -0,0 +1,91 @@
+FROM golang:1.12 as build_tools
+
+# Pinned versions of stuff we pull in
+ENV PROTOBUF_VERSION=3.6.1
+ENV GOGO_PROTOBUF_VERSION=v1.2.2
+ENV GOLANG_PROTOBUF_VERSION=v1.3.1
+ENV PROTOLOCK_VERSION=v0.14.0
+ENV PROTOTOOL_VERSION=v1.8.0
+ENV COUNTERFEITER_VERSION=v6.2.2
+ENV GOIMPORTS_VERSION=379209517ffe
+ENV GOLANGCI_LINT_VERSION=v1.16.0
+
+# Update distro and install dependencies
+RUN apt-get update
```

Comment:
See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/ and combine this line with the next one.

Not sure why the Dockerfile linter isn't getting run against this.

Reply:
Yes, the docker file linter should be getting run on this. I'll update the CI stuff.

Reply:
Delete which cache?

Reply:
Adding the ` && rm -rf /var/lib/apt/lists/*` at the end of the apt-get command, so like:
```
RUN apt-get update && apt-get install -y \
    aufs-tools \
...
    s3cmd=1.1.* \
 && rm -rf /var/lib/apt/lists/*
```

Reply:
Done